network/builder/src/builder.rs

```diff
@@ -170,8 +170,8 @@ impl NetworkBuilder {
         }
     }
 
-    pub fn network_context(&self) -> &NetworkContext {
-        &self.network_context
+    pub fn network_context(&self) -> Arc<NetworkContext> {
+        self.network_context.clone()
```

Comment:
Hehe, I remember I was asked specifically to change this to how it was.  Makes sense this way